dpctl/tensor/libtensor/source/full_ctor.cpp

```diff
@@ -194,8 +249,42 @@ usm_ndarray_full(const py::object &py_value,
             full_contig_event);
     }
     else {
-        throw std::runtime_error(
-            "Only population of contiguous usm_ndarray objects is supported.");
+        int nd = dst.get_ndim();
+        auto const &dst_shape = dst.get_shape_vector();
+        auto const &dst_strides = dst.get_strides_vector();
+
+        auto fn = full_strided_dispatch_vector[dst_typeid];
```

Comment:
At some point it makes sense to implement stride simplification logic here too, perhaps in a future pr.

Reply:
Agreed, I think it would have to use `compact_iteration_space` rather than `simplify_iteration_space_1`.